cpp/src/arrow/compute/exec/expression_test.cc

```diff
@@ -1376,12 +1376,18 @@ TEST(Expression, SerializationRoundTrips) {
 
   ExpectRoundTrips(field_ref("field"));
 
+  ExpectRoundTrips(field_ref(FieldRef("foo", "bar", "baz")));
```

Comment:
Do we have to spell out `field_ref(FieldRef("foo", "bar"))` explicitly or is it possible to simply write `field_ref("foo", "bar")`?

Reply:
Currently it doesn't work, `FieldRef("foo", "bar")` needs to be spelled out.

<details>
<summary>Try 1: `field_ref("foo", "bar")`</summary>
<pre>
```
/Users/alenkafrim/repos/arrow/cpp/src/arrow/compute/exec/expression_test.cc:1379:20: error: no matching function for call to 'field_ref'
  ExpectRoundTrips(field_ref("foo", "bar", "baz"));
                   ^~~~~~~~~
/Users/alenkafrim/repos/arrow/cpp/src/arrow/compute/exec/expression.h:152:12: note: candidate function not viable: requires single argument 'ref', but 3 arguments were provided
Expression field_ref(FieldRef ref);
```
</pre>
</details>

<details>
<summary>Try 2: `field_ref(("foo", "bar"))`</summary>
<pre>
```
/Users/alenkafrim/repos/arrow/cpp/src/arrow/compute/exec/expression_test.cc:1379:31: warning: expression result unused [-Wunused-value]
  ExpectRoundTrips(field_ref(("foo", "bar", "baz")));
```
</pre>
</details>